src/pages/settings/Payments/ChooseTransferAccountPage.js

```diff
@@ -0,0 +1,103 @@
+import KeyboardAvoidingView from '../../../components/KeyboardAvoidingView/index';
+import PaymentMethodList from './PaymentMethodList';
+import ONYXKEYS from '../../../ONYXKEYS';
+import compose from '../../../libs/compose';
+import * as paymentPropTypes from './paymentPropTypes';
+import * as PaymentMethods from '../../../libs/actions/PaymentMethods';
+import CONST from '../../../CONST';
+
+const propTypes = {
+    walletTransfer: paymentPropTypes.walletTransferPropTypes,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    walletTransfer: {},
+};
+
+class ChooseTransferAccountPage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.paymentMethodSelected = this.paymentMethodSelected.bind(this);
+    }
+
+    /**
+     * Navigate to the appropriate payment method type addition screen
+     * @param {String} paymentMethodType
+     */
+    navigateToAddPaymentMethod(paymentMethodType) {
+        if (paymentMethodType === CONST.PAYMENT_METHODS.PAYPAL) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_PAYPAL_ME);
+        }
+
+        if (paymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_DEBIT_CARD);
+        }
+
+        if (paymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_BANK_ACCOUNT);
+        }
+    }
+
+    /**
+     * Go back to TransferPage with the selected bank account
+     *
+     * @param {Object} nativeEvent
+     * @param {String} accountID id of the selected account.
+     */
+    paymentMethodSelected(nativeEvent, accountID) {
+        // If accountID is undefined it means that addPaymentMethod button is pressed.
+        if (!accountID) {
+            this.navigateToAddPaymentMethod(this.props.walletTransfer.filterPaymentMethodType);
+            return;
+        }
+        PaymentMethods.updateWalletTransferData({selectedAccountID: accountID});
+        Navigation.navigate(ROUTES.SETTINGS_TRANSFER_BALANCE);
+    }
+
+    render() {
+        return (
+            <ScreenWrapper>
+                <KeyboardAvoidingView>
+                    <HeaderWithCloseButton
+                        title={this.props.translate('chooseTransferAccountPage.chooseAccount')}
+                        shouldShowBackButton
+                        onBackButtonPress={() => Navigation.goBack()}
+                        onCloseButtonPress={() => Navigation.dismissModal()}
+                    />
+                    <View style={[styles.flex1, styles.pv5]}>
+                        <PaymentMethodList
+                            onPress={this.paymentMethodSelected}
+                            enableSelection
+                            filterType={this.props.walletTransfer.filterPaymentMethodType}
+                            selectedAccountID={this.props.walletTransfer.selectedAccountID}
+                        />
```

Comment:
Shouldn't there be an `+ Add Payment Method` button next to this list?

Reply:
That button is handled inside the `paymentMethodList`

Reply:
Yes it's duplicating logic see the comment below this one 😄

Reply:
It is not completely duplicating the code. It is just changing the title of that button based on the payment method and we can directly use that button to navigate to the adding selected payment method type. I don't think it is good to show the popover menu if there is always going to be one menu item.

And again, the same component is used on that page.

Reply:
Sure, I will do that change. 
I will move the `+ Add Payment Method` button to that component and use it. I think it would make more sense then. and filter it based on prop.